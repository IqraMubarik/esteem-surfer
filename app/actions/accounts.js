import { encryptKey } from '../utils/crypto';
import { setItem, removeItem } from '../helpers/storage';

export const ACCOUNT_ADDED = 'accounts/ACCOUNT_ADDED';
export const ACCOUNT_DELETED = 'accounts/ACCOUNT_DELETED';
export const ACCOUNT_ACTIVATED = 'accounts/ACCOUNT_ACTIVATED';

export const addAccount = (username, keys) => (dispatch, getState) => {
  const { global } = getState();

  const { pin } = global;

  const eKeys = Object.assign(
    {},
    ...Object.keys(keys).map(k => ({ [k]: encryptKey(keys[k], pin) }))
  );

  const accountData = {
    type: 's',
    username,
    keys: eKeys
  };

  setItem(`account_${username}`, accountData);
  setItem(`active_account`, username);
  dispatch(accountAdded(username, accountData));
};

export const deleteAccount = username => (dispatch, getState) => {
  removeItem(`account_${username}`);

  const { accounts } = getState();

  if (accounts.active === username) {
    setItem(`active_account`, null);
  }

  accountDeleted(username);
};

export const activateAccount = username => dispatch => {
  setItem(`active_account`, username);
  dispatch(accountActivated(username));
};

/* action creators */

export const accountAdded = username => ({
  type: ACCOUNT_ADDED,
  payload: { username }
});

export const accountDeleted = username => ({
  type: ACCOUNT_DELETED,
  payload: { username }
});

export const accountActivated = username => ({
  type: ACCOUNT_ACTIVATED,
  payload: { username }
});
