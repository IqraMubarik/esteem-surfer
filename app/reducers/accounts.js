import { getByPrefix, getItem } from '../helpers/storage';

import {
  ACCOUNT_ADDED,
  ACCOUNT_DELETED,
  ACCOUNT_ACTIVATED,
  ACCOUNT_ADDED_SC,
  ACCOUNT_DEACTIVATED,
  ACTIVE_ACCOUNT_DATA_UPDATED
} from '../actions/accounts';

const defaultState = {
  activeAccount: getItem('active_account')
    ? Object.assign(
        {},
        { accountData: null },
        getByPrefix('account_').filter(
          i => i.username === getItem('active_account')
        )[0]
      )
    : null,
  list: getByPrefix('account_')
};

export default function accounts(state = defaultState, action) {
  switch (action.type) {
    case ACCOUNT_ADDED:
    case ACCOUNT_ADDED_SC:
    case ACCOUNT_DELETED:
    case ACCOUNT_ACTIVATED:
    case ACCOUNT_DEACTIVATED: {
      return {
        activeAccount: getItem('active_account')
          ? Object.assign(
              {},
              { accountData: null },
              getByPrefix('account_').filter(
                i => i.username === getItem('active_account')
              )[0]
            )
          : null,
        list: getByPrefix('account_')
      };
    }
    case ACTIVE_ACCOUNT_DATA_UPDATED: {
      const { accountData } = action.payload;
      return {
        activeAccount: Object.assign(
          {},
          { accountData },
          getByPrefix('account_').filter(
            i => i.username === getItem('active_account')
          )[0]
        ),
        list: getByPrefix('account_')
      };
    }
    default:
      return state;
  }
}
