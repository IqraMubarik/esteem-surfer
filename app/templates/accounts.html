<nav-bar></nav-bar>
<div class="app-container" id="content-main">
  <div class="app-content dynamic-page accounts-page">
    <h1 class="page-title">ACCOUNTS</h1>

    <table class="table table-striped accounts-list">
      <tbody>
      <tr ng-repeat="account in accounts">
        <td style="width: 30px">
          <img class="login-type-image" ng-if="account.type==='s'" src="img/steem.svg" title="Steem">
          <img class="login-type-image" ng-if="account.type==='sc'" src="img/steem-connect-logo.svg" title="Steem Connect">
        </td>
        <td style="width: 30px"><div class="profile-image" author-bg-img-style author="{{ account.username }}"> </div> </td>
        <td>@{{ account.username }} <span class="active" nf-if="activeUsername === account.username">{{ 'ACTIVE' | translate }}</span></td>
        <td style="width: 30px"><button class="btn btn-primary btn-sm" ng-disabled="activeUsername === account.username">Login</button> </td>
        <td style="width: 30px"><button class="btn btn-danger btn-sm" ng-disabled="activeUsername === account.username">Remove</button> </td>
      </tr>
      </tbody>
    </table>
  </div>
</div>
<app-footer></app-footer>

