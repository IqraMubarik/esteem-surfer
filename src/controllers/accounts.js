export default ($scope, $rootScope, userService, activeUsername) => {

  const loadData = () => {
    $scope.accounts = userService.getAll();
    $scope.activeUsername = activeUsername();
  };

  loadData();
};
