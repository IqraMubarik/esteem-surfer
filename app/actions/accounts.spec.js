/* eslint-disable */
import { spy } from 'sinon';
import * as actions from './accounts';

describe('accounts actions', () => {
  it('(1) accountAdded action creator', () => {
    expect(actions.accountAdded('foo')).toMatchSnapshot();
  });

  it('(2) accountDeleted action creator', () => {
    expect(actions.accountDeleted('foo')).toMatchSnapshot();
  });

  it('(3) accountActivated action creator', () => {
    expect(actions.accountActivated('foo')).toMatchSnapshot();
  });
});
