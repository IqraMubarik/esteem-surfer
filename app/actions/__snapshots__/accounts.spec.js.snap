// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounts actions (1) accountAdded action creator 1`] = `
Object {
  "payload": Object {
    "username": "foo",
  },
  "type": "accounts/ACCOUNT_ADDED",
}
`;

exports[`accounts actions (2) accountDeleted action creator 1`] = `
Object {
  "payload": Object {
    "username": "foo",
  },
  "type": "accounts/ACCOUNT_DELETED",
}
`;

exports[`accounts actions (3) accountActivated action creator 1`] = `
Object {
  "payload": Object {
    "username": "foo",
  },
  "type": "accounts/ACCOUNT_ACTIVATED",
}
`;
